### PR TITLE
Instructions for Perforce SSO using Google Identity and OIDC

### PR DESCRIPTION
**Pull request by davidair**

### CODE REVIEW

File: google_sso/README.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+In the Google Cloud Console, navigate to API & Services and select Credentials.
+Click on Create Credentials, select "OAuth client ID" and pick "Web application"
+as the Application type:
+
+![Create Credentials](create_credentials.png)
+
+You can name the credentials "HAS Client" (the name does not matter).
+
+Note that you will need to provide the authorized Authorized redirect URIs.
+Since we are not using a domain name but an IP address, we would need to use a special domain,
```
Comment (noahdungey) [inline google_sso/README.md:69]: Add a recommendation that users should use their own domainname for a production environment.

File: google_sso/README.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+**Please consult a security or network engineer when adapting this to your environment.**
+
+To open up all VMs on your GCP network to incoming traffic on ports 1666 and 3000, follow these steps:
+
+-  In Google Cloud Console, navigate to VPC Network / Firewall
+-  Click on "Create Firewall Rule"
+
+![Create Firewall](create_firewall.png)
+
+You can give your rule any name (e.g. "helix-and-has"). Select "All instances in the network"
```
Comment (noahdungey) [inline google_sso/README.md]: Two separate firewall rules, one for Helix (1666/tcp) and another for HAS (3000/tcp) are more appropriate use of least privilege principle.

File: google_sso/README.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+#### Installing and configuring Helix Core
+
+The full guide on installing Helix Core is available here:
+https://www.perforce.com/manuals/p4sag/Content/P4SAG/install.linux.packages.install.html
+
+In a nutshell:
+
+```
+wget -qO - https://package.perforce.com/perforce.pubkey | sudo apt-key add -
+sudo sh -c "echo 'deb http://package.perforce.com/apt/ubuntu xenial release' > /etc/apt/sources.list.d/perforce.list"
```
Comment (noahdungey) [inline google_sso/README.md]: Better to use sudo tee then sudo sh.
`echo 'deb http://package.perforce.com/apt/ubuntu xenial release' | sudo tee /etc/apt/sources.list.d/perforce.list`

File: google_sso/README.md [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+-  URI for OIDC issuer: https://accounts.google.com
+-  OIDC client ID: CLIENT_ID_SAVED_EARLIER
+-  OIDC client secret: CLIENT_SECRET_SAVED_EARLIER
+
+Answer "Yes" to "Do you wish to continue?"
+
+To test that the server is up and running, you can navigate to https://VM_EXTERNAL_IP.xip.io:3000/.
+
+Please note that your browser may block this because the site is using a default self-signed
+TSL certificate. In Chrome, you can click on "Advanced" when this happens to proceed.
```
Comment (noahdungey) [inline google_sso/README.md]: typo: s/TSL/TLS/

File: google_sso/README.md [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+p4 -u testuser -p ssl:VM_EXTERNAL_IP:1666 depots
+```
+
+## Important considerations
+
+This guide has been written with the focus on quick demonstration and minimum prerequisites.
+However, a lot of shortcuts taken here should not be used in a production environment.
+Here is a list of important considerations:
+
+## TSL Certificates
```
Comment (noahdungey) [inline google_sso/README.md]: s/TSL/TLS/

**Review by noahdungey — CHANGES_REQUESTED** [review]
A few nits.

File: google_sso/README.md (hunk 1 above)
Comment (davidair) [reply google_sso/README.md:69]: Added the recommendation

File: google_sso/README.md (hunk 5 above)
Comment (davidair) [reply google_sso/README.md]: Done

File: google_sso/README.md (hunk 2 above)
Comment (davidair) [reply google_sso/README.md]: Done

**Comment by davidair** [comment]
Addressed all the comments.

**Review by noahdungey — APPROVED** [review]
While I would prefer that the two firewall rules be set by network tag and then both "helix" and "has" tags be added to your instance. That can be left for a future change.

**Pull request merged**